Svc/FpySequencer/FpySequencerDirectives.cpp

```diff
@@ -423,55 +426,72 @@ Signal FpySequencer::binaryCmp_directiveHandler(const FpySequencer_BinaryCmpDire
         return Signal::stmtResponse_success;
     }
 
-    // okay, it is an inequality comparison
-
-    // whether the comparison is signed or unsigned
-    bool sign = true;
-
-    if (directive.get_op() >= Fpy::DirectiveId::ULT && directive.get_op() <= Fpy::DirectiveId::UGE) {
-        sign = false;
-    }
-
     I8 cmpResult;
 
-    if (sign) {
-        cmpResult = (lhs == rhs) ? 0 : (lhs < rhs) ? -1 : 1;
+    if (directive.get_op() >= Fpy::DirectiveId::FEQ && directive.get_op() <= Fpy::DirectiveId::FGE) {
+        F64 frhs, flhs;
+        memcpy(&frhs, &rhs, sizeof(frhs));
+        memcpy(&flhs, &lhs, sizeof(flhs));
+
+        if (std::isunordered(flhs, frhs)) {
+            // nan is one of the args
+            // always fail
+            res = 0;
+            return Signal::stmtResponse_success;
+        } else if (std::isgreater(flhs, frhs)) {
+            cmpResult = 1;
+        } else if (std::isless(flhs, frhs)) {
+            cmpResult = -1;
+        } else {
+            cmpResult = 0;
+        }
+
+        // handle the floating equality cmps
+        if (directive.get_op() == Fpy::DirectiveId::FEQ) {
+            res = cmpResult == 0;
+            return Signal::stmtResponse_success;
+        }
+        if (directive.get_op() == Fpy::DirectiveId::FNE) {
+            res = cmpResult != 0;
+            return Signal::stmtResponse_success;
+        }
     } else {
-        // unsigned comparison. static cast to unsigned longs
-        U64 ulhs = static_cast<U64>(lhs);
-        U64 urhs = static_cast<U64>(rhs);
-        cmpResult = (ulhs == urhs) ? 0 : (ulhs < urhs) ? -1 : 1;
+        // integer comparison
+        if (directive.get_op() >= Fpy::DirectiveId::ULT && directive.get_op() <= Fpy::DirectiveId::UGE) {
+            // unsigned comparison. static cast to unsigned longs
+            U64 ulhs = static_cast<U64>(lhs);
+            U64 urhs = static_cast<U64>(rhs);
+            cmpResult = (ulhs == urhs) ? 0 : (ulhs < urhs) ? -1 : 1;
+        } else {
+            cmpResult = (lhs == rhs) ? 0 : (lhs < rhs) ? -1 : 1;
+        }
     }
 
     if (cmpResult == 0) {
         // values were equal
         // result is true if equality is okay
-        res = (directive.get_op() == Fpy::DirectiveId::UGE 
-            || directive.get_op() == Fpy::DirectiveId::ULE 
-            || directive.get_op() == Fpy::DirectiveId::SGE 
-            || directive.get_op() == Fpy::DirectiveId::SLE);
+        res = (directive.get_op() == Fpy::DirectiveId::UGE || directive.get_op() == Fpy::DirectiveId::ULE ||
+               directive.get_op() == Fpy::DirectiveId::SGE || directive.get_op() == Fpy::DirectiveId::SLE ||
+               directive.get_op() == Fpy::DirectiveId::FGE || directive.get_op() == Fpy::DirectiveId::FLE);
```

Comment:
You've already handled `FEQ` and `FNE` above, why handle the GE and LE cases here?